controllers/operands/networkAddons_test.go

```diff
@@ -507,6 +507,58 @@ var _ = Describe("CNA Operand", func() {
 			}),
 		)
 
+		type ksdAnnotationParams struct {
+			annotationExists  bool
+			annotationValue   string
+			ksdDeployExpected bool
+		}
+		DescribeTable("when reconciling kube-secondary-dns", func(o ksdAnnotationParams) {
+			hcoKSDConfig := commonTestUtils.NewHco()
+			hcoKSDConfig.Annotations = map[string]string{}
+
+			if o.annotationExists {
+				hcoKSDConfig.Annotations["deployKSD"] = o.annotationValue
+			}
+
+			existingResource, err := NewNetworkAddons(hcoKSDConfig)
+			Expect(err).ToNot(HaveOccurred())
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Err).ToNot(HaveOccurred())
+
+			foundResource := &networkaddonsv1.NetworkAddonsConfig{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+					foundResource),
+			).ToNot(HaveOccurred())
+
+			if o.ksdDeployExpected {
+				Expect(existingResource.Spec.KubeSecondaryDNS).ToNot(BeNil(), "KSD spec should be added")
+			} else {
+				Expect(existingResource.Spec.KubeSecondaryDNS).To(BeNil(), "KSD spec should not be added")
+			}
+		},
+			Entry("should have ovs if deployKSD annotation is set to true", ksdAnnotationParams{
+				annotationExists:  true,
+				annotationValue:   "true",
+				ksdDeployExpected: true,
+			}),
+			Entry("should not have ovs if deployKSD annotation is not set to true", ksdAnnotationParams{
+				annotationExists:  true,
+				annotationValue:   "false",
+				ksdDeployExpected: false,
+			}),
+			Entry("should not have ovs if deployKSD annotation does not exist", ksdAnnotationParams{
+				annotationExists:  false,
+				annotationValue:   "",
+				ksdDeployExpected: false,
+			}),
```

Comment:
Missing test case? When `desplayKSD` is *not* set, but now it change to `"true"`. Is it a valid case? I know it can't be tested by the current test function.

Reply:
if the annotation doesnt exists it can't be set true
please see the test itself
it is not about upgrade, it is about if to deploy or not deploy the component in the current version

Reply:
I thought about a test like this. WDYT?

1. create NetworkAddons object from `hco`: 
    ```go
    existingResource, err := NewNetworkAddons(hco)
    ```
2. check the created object
    ```go
    Expect(existingResource.Spec.KubeSecondaryDNS).To(BeNil(), "KSD spec should not be added")
    ```
3. modify `hco` to add the annotation
4. initialize the client with the "old" NetworkAddons and the modified hco.
5. run `handler.ensure(req)`
6. check the result - 
    ```go
    Expect(existingResource.Spec.KubeSecondaryDNS).To(BeNil(), "KSD spec should not be added")
    ```

Reply:
We can just add to the first case that it will check it is not on the CR before it adds it right ?

Reply:
I mean the use case where the NetworkAddons is w/o the `Spec.KubeSecondaryDNS` field, and the HC is w/o the annotation, and the user now sets the annotation. I don't think its covered.

Reply:
If we break down this secanario, case 2 and 3 are checking the first part of the sentence
"NetworkAddons is w/o the Spec.KubeSecondaryDNS field, and the HC is w/o the annotation"
and then case 1 checks that if you add it, it is deployed
do you agree?

I don't think we need another one that check both, as the building blocks are checked

Reply:
Thanks, as we spoke once https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2191 is merged
Will adapt this PR